src/System.Text.RegularExpressions/src/System/Text/RegularExpressions/RegexInterpreter.cs

```diff
@@ -964,8 +965,18 @@ protected override void Go()
                             string set = _code.Strings[Operand(0)];
 
                             while (c-- > 0)
+                            {
+                                // Check the timeout every 2000th iteration. The aditional if check
+                                // in every iteration can be neglected as the cost of the CharInClass
+                                // check is many times higher.
+                                if (c % LoopTimeoutCheckCount == 0)
```

Comment:
The codegen here would be better if a) LoopTimeoutCheckCount was a power of 2, and b) if c was cast to uint:
https://sharplab.io/#v2:EYLgxg9gTgpgtADwGwBYA0AXEBDAzgWwB8ABAJgEYBYAKGIGYACMhgYQYG8aHunHgIIAGwYBZcgAoAlgDsMDGQAcArhgCUDALwA+edOVyApA1IAGM5o0MTAbi496DfkNGkps3fvXaPKhkdMoABwWVrbUPLyOAsIidG5yiipeOuJKMmqJhsYmQSE2NAC+QA==